### PR TITLE
Added Test Cases

### PR DESCRIPTION
**Pull request by MyNewGitHere**
Added test cases for 
app.tsx
footer-comp.tsx
menu-header.tsx
site-name-header.tsx
load-more-button.tsx, 
news-card.tsx

### CODE REVIEW

**Comment by MyNewGitHere** [comment]
Addressed previous issues like :

Use data-testid and getByTestId for elements as much as possible
Use assertions to cover each and every statement(tried for max)
Try to reach maximum test coverage

File: package.json [hunk 1]
```diff
@@ -8,11 +8,6 @@
     "@testing-library/jest-dom": "^5.17.0",
     "@testing-library/react": "^13.4.0",
     "@testing-library/user-event": "^13.5.0",
```
Comment (aashitk) [inline package.json]: This can be in devDependencies

File: src/App.test.tsx
Comment (aashitk) [inline src/App.test.tsx:1]: News card component is mocked but not tested.

File: src/components/menu-header.test.tsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    const { getByTestId } = render(
+      <MenuHeader
+        setApi={setApiMock}
+        setStart={setStartMock}
+        setEnd={setEndMock}
+      />
+    );
+
+    const newButton = getByTestId("new-button");
+    const pastButton = getByTestId("past-button");
```
Comment (aashitk) [inline src/components/menu-header.test.tsx]: Should add assertions for this components below

File: src/components/footer-comp.test.tsx
Comment (aashitk) [inline src/components/footer-comp.test.tsx:1]: You need to mock AppBar and Toolbar, and test it using assertions

File: src/components/footer-comp.test.tsx [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import { render } from "@testing-library/react";
+import Footer from "./footer-comp";
+import footerLogo from "../asset/hackernews-footer.svg";
+
+describe("Footer component", () => {
+  test("renders the logo image", () => {
+    const { getByAltText } = render(<Footer />);
+    const logoImage = getByAltText('Vector');
```
Comment (aashitk) [inline src/components/footer-comp.test.tsx]: Since data-testid is present, why not use it to get element

File: src/components/news-card.test.tsx [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    expect(descriptionElement).toBeInTheDocument();
+
+    const metadataElement = getByTestId('news-metadata');
+    expect(metadataElement).toBeInTheDocument();
+
+    const expectedTimestampRegex = /\d{1,2}\/\d{1,2}\/\d{4}, \d{1,2}:\d{2}:\d{2} (AM|PM)/;
+    expect(metadataElement.textContent).toMatch(expectedTimestampRegex);
+
+    const expectedScoreRegex = new RegExp(`${mockNews.score} Comments`);
+    expect(metadataElement.textContent).toMatch(expectedScoreRegex);
```
Comment (aashitk) [inline src/components/news-card.test.tsx]: Do not use RegExp, check exact value that is expected to show up

File: src/components/site-name-header.test.tsx [hunk 5]
```diff
@@ -0,0 +1,17 @@
+import { render } from '@testing-library/react';
+import SiteNameHeader from './site-name-header';
+
+describe('SiteNameHeader component', () => {
+  it('renders the header with the correct logo image', () => {
+    const { getByAltText } = render(<SiteNameHeader />);
+    const logoImage = getByAltText('headerlogo');
```
Comment (aashitk) [inline src/components/site-name-header.test.tsx]: Why not have data-testid like footer component?

**Review by aashitk — CHANGES_REQUESTED** [review]
@Vebhavdaa Added few change requests.
Make sure to make similar changes in other places as well.

File: src/App.tsx [hunk 6]
```diff
@@ -1,74 +1,23 @@
...
-          ).then((response) => response.json())
-        )
-    )
-      .then((data) => setData(data))
-      .catch((error) => console.error("Error fetching data:", error));
-  }, [ids, start, end]);
-
   return (
-    <>
+    <AppContextProvider>
```
Comment (aashitk) [inline src/App.tsx]: Put context provider out of App.tsx. It's a good practice.

File: src/AppContext.test.tsx [hunk 7]
```diff
@@ -0,0 +1,215 @@
...
+    };
+
+    const { getByTestId } = render(
+      <AppContextProvider>
+        <ContextChecker />
+      </AppContextProvider>
+    );
+
+    const contextCheck = getByTestId("context-check");
+    expect(contextCheck).toBeInTheDocument();
```
Comment (aashitk) [inline src/AppContext.test.tsx:48]: Should also check the text being rendered.

**Review by aashitk — CHANGES_REQUESTED** [review]
Minor change requested

**Review by aashitk — APPROVED** [review]

**Pull request merged**